include/seqan3/alignment/matrix/edit_distance_trace_matrix_full.hpp

```diff
@@ -216,4 +249,126 @@ class edit_distance_trace_matrix_full
     std::vector<column_type> columns{};
 };
 
+template <typename word_t, bool is_semi_global, bool use_max_errors>
+struct edit_distance_trace_matrix_full<word_t, is_semi_global, use_max_errors>::trace_path_iterator
+{
+    /*!\name Associated types
+     * \{
+     */
+    //!\brief Input iterator tag.
+    using iterator_category = std::input_iterator_tag;
+    //!\copydoc seqan3::detail::trace_iterator_base::value_type
+    using value_type = detail::trace_directions;
+    //!\copydoc seqan3::detail::trace_iterator_base::difference_type
+    using difference_type = std::ptrdiff_t;
+    //!\}
+
+    //!\brief Shortcut for seqan3::detail::trace_directions::diagonal;
```

Comment:
Just a question: Why do you put the semicolon in the end of all the comments here ? Since it is consistent I think it has a reason right ?

Reply:
good catch xD